Tests/OpenCombineTests/PublisherTests/DecodeTests.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  DecodeTests.swift
+//  
+//
+//  Created by Joseph Spadafora on 6/21/19.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, *)
+final class DecodeTests: XCTestCase {
+    static let allTests = [
+        ("testDecodeWorks", testDecodeWorks),
+        ("testDownstraemReceivesFailure", testDownstreamReceivesFailure)
+    ]
+
+    private let jsonEncoder = JSONEncoder()
+    private let jsonDecoder = JSONDecoder()
+
+    let testValue = TestDecodable()
+
+    func testDecodeWorks() {
+        let data = try! jsonEncoder.encode(testValue)
+
+        var decodedValue: TestDecodable?
+        _ = Publishers
+            .Just(data)
+            .decode(type: TestDecodable.self, decoder: jsonDecoder)
+            .sink(receiveValue: { foundValue in
```

Comment:
I think we shouldn't use unrelated entities like `Sink` or `Just` unless there's a real need to do so.

You probably better go with `PassthroughSubject`/`CustomPublisher` and `TrackingSubscriber`, because these are customizable, and allow better test isolation: if there's a regression in `Sink` implementation, we don't want the `Decode` tests to fail as well.